iceoryx_examples/icecubetray/iox_cheeky_app.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/publisher.hpp"
+#include "iceoryx_posh/popo/subscriber.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+constexpr char APP_NAME[] = "iox-cpp-cheeky";
+
+int main()
+{
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // When starting this app with the user 'notallowed'
+
+    // 1) Subscribers can be created without any readable shared memory segment, in this case no data will ever arrive
+    iox::popo::Subscriber<RadarObject> subscriber({"Radar", "FrontLeft", "Object"});
+
+    // 2) The publisher object can't be initalised correctly because 'notallowed' does not have write access
```

Comment:
Please describe what happens here. How does the user know that it couldn't be initialized correctly as far as I know the publisher object is always in a valid and initialized state.